packages/notebook/src/widget.ts

```diff
@@ -837,6 +843,13 @@ class Notebook extends StaticNotebook {
       newValue = Math.max(newValue, 0);
       newValue = Math.min(newValue, this.model.cells.length - 1);
     }
+
+    // Blur the previous active cell.
+    let activeCell = this.activeCell;
+    if (activeCell && newValue !== oldValue) {
+      activeCell.editor.blur();
```

Comment:
I think this should be merged into the check below, around line 857, where _activeCell is actually set to a new value.